File: python/src/math/math.py
class Math(object):

    def __init__(self) -> object: pass

    def fibonacci(self, num: int) -> int:
        if not isinstance(num, int):
            return -1
        
        if num <= 0:
            return 0
        
        if num == 1:
            return 1
        
        fibonacci: list = [0,1]
        for i in range(2, num + 1):
            sum = fibonacci[i - 1] + fibonacci[i - 2]
            fibonacci.append(sum)
        
        return fibonacci[-1]
    
    def square(self, num: int) -> int:
        if not isinstance(num, int):
            return -1
        
        return num * num

File: java/src/main/java/learn/source/code/documentation/math/Math.java
package learn.source.code.documentation.math;

// std
import java.util.ArrayList;


public class Math {

    public Math() {}

    public long fibonacci(int num) {
        if (num <= 0) return 0;
        if (num == 1) return 1;

        ArrayList<Long> fibonacci = new ArrayList<Long>();
        fibonacci.add(0L);
        fibonacci.add(1L);

        for (int i = 2; i <= num; i++) {
            try {
                long sum = fibonacci.get(i - 1) + fibonacci.get(i - 2);
                fibonacci.add(sum);
            } catch (Exception ex) {
                ex.printStackTrace();
            }
        }
        return fibonacci.get(fibonacci.size() - 1);
    }

    public long square(int num) {
        return num * num;
    }
}
